mobile/src/lib/background.js

```diff
@@ -0,0 +1,72 @@
+import { getStore } from './store'
+import { initService } from './init'
+import { logException } from './crash-reporter'
+import { loadState } from './localStorage'
+import { startMediaUpload, mediaBackup, endMediaUpload } from '../actions/mediaBackup'
+import { backupAllowed } from './network'
+import { initPolyglot } from '../../../src/lib/I18n'
+
+const hasIosCordovaPlugin = () => {
+  return window.cordova !== undefined && window.cordova.platformId === 'ios' && window.BackgroundFetch !== undefined
+}
+
+export const startBackgroundService = () => {
+  if (hasIosCordovaPlugin()) {
+    startIosBackgroundService()
+  }
+}
+
+export const stopBackgroundService = () => {
+  if (hasIosCordovaPlugin()) {
+    stopIosBackgroundService()
+  }
+}
+
+const startIosBackgroundService = () => {
+  let Fetcher = window.BackgroundFetch
+
+  const fetchCallback = () => {
+    console.log('[js] BackgroundFetch initiated')
+
+    loadState().then(persistedState => {
+      let store = getStore(persistedState)
+      initService(store)
+
+      logException('C\'est moi le Service Background!!!')
+
+      if (store.getState().mobile.settings.backupImages && backupAllowed(store.getState().mobile.settings.wifiOnly)) {
```

Comment:
If you request twice `store.getState()` don't it be reasonable to use a variable?
```js
const state = store.getState()
if (state.mobile.settings.backupImages && backupAllowed(state.mobile.settings.wifiOnly)) { 
// etc.
```